component-macro: Dependency conflict with wit-bindgen causes compilation failure

Hi!

My project depends on both `wit-bindgen` and `wasmtime-component-macro`. I'm grabbing these dependencies from the git repositories directly.

Because of https://github.com/bytecodealliance/wit-bindgen/pull/442, `wit-bindgen` uses `wit-parser` v0.3.1. This is an issue because `wasmtime-component-macro` depends on v0.3, which has API mismatch. Specifically, this error is encountered during compilation:

```
error[E0599]: no function or associated item named `parse` found for struct `World` in the current scope
   --> /Users/amitp/.cargo/git/checkouts/wasmtime-41807828cb3a7a7e/29b23d4/crates/component-macro/src/bindgen.rs:123:24
    |
123 |                 World::parse("<macro-input>", &s.value()).map_err(|e| Error::new(s.span(), e))?;
    |                        ^^^^^
    |                        |
    |                        function or associated item not found in `World`
    |                        help: there is an associated function with a similar name: `parse_file`
```

It seems like a relatively easy fix: https://github.com/bytecodealliance/wit-bindgen/pull/442/files#diff-775b613a11243734c7a6424be2eba37d42f92e2b331001ae06eeb4c65310a5feR152-R154

I would do this myself, but getting set up with the wasmtime repository seems to be a pretty big undertaking and I simply don't have the time.

--- a/crates/component-macro/src/bindgen.rs
+++ b/crates/component-macro/src/bindgen.rs
@@ -5,7 +5,7 @@ use syn::punctuated::Punctuated;
 use syn::token;
 use syn::Token;
 use wasmtime_wit_bindgen::Opts;
-use wit_parser::World;
+use wit_parser::{Document, World};
 
 #[derive(Default)]
 pub struct Config {
@@ -119,8 +119,10 @@ impl Parse for Opt {
             let span = input.parse::<kw::inline>()?.span;
             input.parse::<Token![:]>()?;
             let s = input.parse::<syn::LitStr>()?;
-            let world =
-                World::parse("<macro-input>", &s.value()).map_err(|e| Error::new(s.span(), e))?;
+            let world = Document::parse("<macro-input>".as_ref(), &s.value())
+                .map_err(|e| Error::new(s.span(), e))?
+                .into_world()
+                .map_err(|e| Error::new(s.span(), e))?;
             Ok(Opt::Inline(span, world))
         } else if l.peek(kw::tracing) {
             input.parse::<kw::tracing>()?;

--- a/supply-chain/audits.toml
+++ b/supply-chain/audits.toml
@@ -998,3 +998,9 @@ criteria = "safe-to-deploy"
 version = "0.3.0"
 notes = "The Bytecode Alliance is the author of this crate."
 
+[[audits.wit-parser]]
+who = "Alex Crichton <alex@alexcrichton.com>"
+criteria = "safe-to-deploy"
+version = "0.3.1"
+notes = "The Bytecode Alliance is the author of this crate."
+

